content/usgs-packages/sbtools_Modify.Rmd

```diff
@@ -0,0 +1,302 @@
+
+```{r sbtools-upload}
+dir <- system.file("examples", package="sbtools")
+sbtools_files <- list.files(dir)[1:2]
+sbtools_files
+sbtools_filepaths <- file.path(dir, sbtools_files)
+test_item <- item_upload_create(parent_id = user_id(),
+                                files = sbtools_filepaths)
+test_item
+```
+
+Currently, the title of the new item defaults to the first file that is uploaded, and there is no argument to override this behavior. This is a known `sbtools` [issue](https://github.com/USGS-R/sbtools/issues/49). For now, you can change the title using the function `item_update`, which will be discusssed later in this lesson. Another interesting behavior is that uploading multiple files at once sometimes does not work for complex files. This is a [known issue](https://github.com/USGS-R/sbtools/issues/39). In the meantime, always verify that your files were actually uploaded.
+
+```{r sbtools-upload-verify}
+is.sbitem(item_get(test_item))
+sbtools_files %in% item_list_files(test_item$id)$fname
+```
+
+Now that you have a ScienceBase item, you can append additional files to it using the `item_append_files` function. Use the third file available in `sbtools` to practice appending files. Then, verify that this new file was added to the existing item.
+
+```{r sbtools-append}
+another_sbtools_file <- list.files(dir)[3]
+another_sbtools_filepath <- file.path(dir, another_sbtools_file)
+item_append_files(sb_id = test_item$id,
+                  files = another_sbtools_filepath)
+
+another_sbtools_file %in% item_list_files(test_item)$fname
+```
+
+## Editing your items
+
+You can move, edit, and update your files all from R using `sbtools` functions. This can be useful if you have edits that can be done in bulk, or want to automate the process. 
+
+This section will give examples for the following functions: 
+
+* `item_update`
+* `items_update`
+* `item_upsert`
+* `items_upsert`
+* `item_update_identifier`
+
+The first four functions have very similar behavior and syntax. The obvious difference is whether the function has `item` or `items` for updating an individual item or multiple items at once. The other is that there is "update" and "upsert". Update functions only work if the item already exists. Alternatively, "upsert" updates an existing item, but creates a new item if it doesn't already exist. So for an existing item, `*_update` and `*_upsert` functions have the same behavior.
+
+To use these functions, you need to provide the `sbitem` or id and a list of the metadata info that you intend to update. The list should be key-value pairs of the metadata field and the corresponding value. These fields can be `title`, `browseCategories`, `contacts`, etc. Additional fields can be found by looking at the [developer documentation for a SB item model](https://my.usgs.gov/confluence/display/sciencebase/ScienceBase+Item+Core+Model).
```

Comment:
probably...I struggled with which article to use in front of "SB". I think technically "a" is correct because the first letter is not a vowel, but if you read "SB" out loud as the letters "S.B.", then "a" sounds funny and I would lean towards using "an". BUT if you read "SB" out loud as "ScienceBase", then "a" should be used. :fist_oncoming: English grammar